share-conversion/src/gf2_128/core/a2m.rs

```diff
@@ -0,0 +1,58 @@
+//! This module implements the A2M algorithm.
+
+use super::MulShare;
+use super::{Gf2_128ShareConvert, OTEnvelope};
+use crate::gf2_128::{inverse, mul};
+use rand::{Rng, SeedableRng};
+use rand_chacha::ChaCha12Rng;
+
+/// An additive share of `A = x + y`
+#[derive(Clone, Copy, Debug)]
+pub struct AddShare(u128);
+
+impl AddShare {
+    /// Turn into a multiplicative share and get values for OT
+    ///
+    /// This function returns
+    ///   * `MulShare` - The sender's multiplicative share
+    ///   * `OTEnvelope` - Used for oblivious transfer
+    pub fn convert_to_multiplicative(&self) -> (MulShare, OTEnvelope) {
+        let mut rng = ChaCha12Rng::from_entropy();
+
+        let random: u128 = rng.gen();
+        if random == 0 {
+            panic!("Random u128 is 0");
+        }
```

Comment:
this will be an issue if `random` is 0, since the receiver will receive 2 equal msgs. The receiver will know that the Sender's share is 0.

Reply:
@themighty1 is this to address a faulty rng implementation? 2^(-128) is equivalent to will never happen in our lifetimes.

Reply:
I see... then I propose the following instead, c.f. https://doc.rust-lang.org/stable/std/num/struct.NonZeroU128.html:
```rust
        // We need to exclude 0 here, because it does not have an inverse
        // which is needed later
        let random: NonZeroU128 = rng.gen();
        let random = random.get();
```